flink-table-store-connector/src/main/java/org/apache/flink/table/store/connector/source/TableStoreSource.java

```diff
@@ -159,14 +158,12 @@ public String asSummaryString() {
 
     @Override
     public Result applyFilters(List<ResolvedExpression> filters) {
-        if (tableStore.partitioned()) {
+        if (logStoreTableFactory == null && tableStore.partitioned()) {
             classifyFilters(filters);
         } else {
             fieldFilters = filters;
         }
-        return Result.of(
-                new ArrayList<>(filters),
-                fieldFilters == null ? Collections.emptyList() : fieldFilters);
+        return Result.of(new ArrayList<>(filters), fieldFilters);
```

Comment:
I think it is better to here: `isStreaming && logStoreTableFactory != null ? filters : fieldFilters`.
- First, I think you lack isStreaming. `logStoreTableFactory` always exists when log.system=kafka.
- Second, Even if there is log system, the case of hybrid needs partition filter to filter filestore a large amount of data, which is very important.

Reply:
> I think it is better to here: isStreaming && logStoreTableFactory != null ? filters : fieldFilters.
> First, I think you lack isStreaming. logStoreTableFactory always exists when log.system=kafka.
Second, Even if there is log system, the case of hybrid needs partition filter to filter filestore a large amount of data, which is very important.

You're right on that. Besides, we should be careful when coping with remaining filters for hybrid read conditions.